sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -165,58 +191,56 @@ rom_error_t otbn_imem_sec_wipe(void) {
   return otbn_cmd_run(kOtbnCmdSecWipeImem, kErrorOtbnSecWipeImemFailed);
 }
 
-rom_error_t otbn_imem_write(uint32_t offset_bytes, const uint32_t *src,
-                            size_t num_words) {
-  HARDENED_RETURN_IF_ERROR(
-      check_offset_len(offset_bytes, num_words, kOtbnIMemSizeBytes));
-  otbn_write(kBase + OTBN_IMEM_REG_OFFSET + offset_bytes, src, num_words);
-  return kErrorOk;
-}
-
 rom_error_t otbn_dmem_sec_wipe(void) {
   return otbn_cmd_run(kOtbnCmdSecWipeDmem, kErrorOtbnSecWipeDmemFailed);
 }
 
-rom_error_t otbn_dmem_write(uint32_t offset_bytes, const uint32_t *src,
-                            size_t num_words) {
-  HARDENED_RETURN_IF_ERROR(
-      check_offset_len(offset_bytes, num_words, kOtbnDMemSizeBytes));
-  otbn_write(kBase + OTBN_DMEM_REG_OFFSET + offset_bytes, src, num_words);
-  return kErrorOk;
+/**
+ * Checks if the OTBN application's IMEM and DMEM address parameters are valid.
+ *
+ * IMEM and DMEM ranges must not be "backwards" in memory, with the end address
+ * coming before the start address, and the IMEM range must additionally be
+ * non-empty. Finally, separate sections in DMEM must not overlap each other
+ * when converted to DMEM address space.
+ *
+ * @param app the OTBN application to check
+ * @return true if the addresses are valid, otherwise false.
+ */
+static bool check_app_address_ranges(const otbn_app_t *app) {
```

Comment:
* Should return `rom_error_t` instead of `bool`.
* Call sites should be wrapped with `HARDENED_RETURN_IF(...)`.
* Guard `kErrorOk` return with redundant checks of both conditions. At the end of the function return `kError...` instead of `true`.